src/instructlab/model/remove.py

```diff
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+import shutil
+
+# Third Party
+import click
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf, is_model_safetensors
+
+
+class RemovalError(Exception):
+    """Custom exception for removal errors."""
+    pass
+
+
+def remove_item(model_path: Path, model_name: str, item_type: str):
+    """Remove the specified file or directory."""
+    try:
+        if item_type == "file":
+            print(f"Removing model file: {model_name}.")
+            model_path.unlink()
+        elif item_type == "directory":
+            print(f"Removing model directory: {model_name}.")
+            shutil.rmtree(model_path)
+        print(f"Model {model_name} has been removed.")
+    except OSError as e:
+        raise RemovalError(f"Error while trying to remove {model_name}: {e}") from e
+
+
+@click.command(name="remove")
+@clickext.display_params
+@click.option(
+    "-m",
+    "--model",
+    required=True,
+    help="Specify the model name to remove(check with `ilab model list` first), \n"
+         "e.g. merlinite-7b-lab-Q4_K_M.gguf or instructlab/merlinite-7b-pt.",
+)
+@click.option(
+    "-y",
+    "--yes",
```

Comment:
I think when it comes to data removal we need something more robust even though it diverges from what we have in place in other parts of the code. Wdyt?

Reply:
> Personal opinion: If we do want to separate 3 from 2, I don't love --yes-i-really-mean-it for the name because it doesn't specifically address the destructive nature and that you are telling it to skip the prompt. 

Well, if the CLI is called like so `ilab model remove --yes-i-really-mean-it` I think we understand the destructive nature of it, don't we?

Reply:
If you require a human to confirm, I would go with --force to override, not --yes.

Reply:
+1 to --force.  I have seen that pattern used to address cases like this.  It's succinct and I think conveys the intent clearly.

Reply:
I'm ok with --force. Thanks for the suggestion @rhatdan